src/nvim/terminal.c

```diff
@@ -465,12 +453,8 @@ static int terminal_execute(VimState *state, int key)
       break;
 
     case K_EVENT:
-      // We cannot let an event free the terminal yet. It is still needed.
-      s->term->refcount++;
       multiqueue_process_events(main_loop.events);
-      s->term->refcount--;
```

Comment:
Whatever if anything that needs to "hold on" to the terminal should increment the refcount of the channel instead.